### PR TITLE
fix(docs): doc files kebab-case conformance

### PR DESCRIPTION
**Pull request by pdelewski**
##### Description

Fill in your description here.

---

##### Checklist

- [ ] Changelog updated

###### Testing performed

- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
We need to verify if we link to our documentation from outside of this repo. If yes, then we probably need to create symlinks in order to keep compatibility until we fix this extended documentation. I hope there is no such

**Comment by swiatekm-sumo** [comment]
> We need to verify if we link to our documentation from outside of this repo. If yes, then we probably need to create symlinks in order to keep compatibility until we fix this extended documentation. I hope there is no such

We definitely have links to it in help.sumologic.com. For example: https://help.sumologic.com/Observability_Solution/Kubernetes_Solution/01Kubernetes_Quickstart links to https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/docs/Troubleshoot_Collection.md.

So we definitely need to add symlinks to the renamed files right now. In the future, once the migration to Docusaurus is done, it should be much simpler to replace all of these. CC @kimsauce

**Comment by kimsauce** [comment]
> > We need to verify if we link to our documentation from outside of this repo. If yes, then we probably need to create symlinks in order to keep compatibility until we fix this extended documentation. I hope there is no such
> 
> We definitely have links to it in help.sumologic.com. For example: https://help.sumologic.com/Observability_Solution/Kubernetes_Solution/01Kubernetes_Quickstart links to https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/docs/Troubleshoot_Collection.md.
> 
> So we definitely need to add symlinks to the renamed files right now. In the future, once the migration to Docusaurus is done, it should be much simpler to replace all of these. CC @kimsauce

That is a good idea. Let me know if you need anything from us. Our site goes live Sept. 23.

File: ci/markdown_table_formatter.sh [hunk 1]
```diff
@@ -10,8 +10,8 @@ if ! "${CMD}" --version ; then
 fi
 
 readonly FILES="deploy/README.md
-deploy/docs/Best_Practices.md
-deploy/docs/Performance.md
+deploy/docs/best-practices.md
+deploy/docs/performance.md
```
Comment (sumo-drosiek) [inline ci/markdown_table_formatter.sh:14]:
```suggestion
docs/best-practices.md
docs/performance.md
```

File: deploy/helm/sumologic/upgrade-2.0.0.sh [hunk 2]
```diff
@@ -1080,7 +1080,7 @@ function check_fluentd_persistence() {
     warning "This chart starting with 2.0 sets it to 'true' by default"
     warning "We're migrating your values.yaml with persistence set to 'false'"
     warning "Please refer to the following doc in in case you wish to enable it:"
-    warning "https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/docs/Best_Practices.md#fluentd-file-based-buffer"
+    warning "https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/docs/best-practices.md#fluentd-file-based-buffer"
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/upgrade-2.0.0.sh:1083]:
```suggestion
    warning "https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/docs/best-practices.md#fluentd-file-based-buffer"
```

File: deploy/helm/sumologic/values.yaml [hunk 3]
```diff
@@ -506,7 +506,7 @@ fluentd:
         requests:
           memory: 768Mi
           ## Warning! Increasing the CPU requests for Fluentd above 1000m might result in broken autoscaling
-          ## ref: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/docs/Best_Practices.md#cpu-resources-warning
+          ## ref: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/docs/best-practices.md#cpu-resources-warning
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:509]:
```suggestion
          ## ref: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/docs/best-practices.md#cpu-resources-warning
```

File: deploy/helm/sumologic/values.yaml [hunk 4]
```diff
@@ -1215,7 +1215,7 @@ kube-prometheus-stack:
   ## Uncomment the flag below to not install kube-prometheus-stack helm chart
   ## as a dependency along with this helm chart.
   ## This is needed e.g. if you want to use a different version of kube-prometheus-stack -
-  ## see /deploy/docs/Best_Practices.md#using-newer-kube-prometheus-stack.
+  ## see /deploy/docs/best-practices.md#using-newer-kube-prometheus-stack.
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:1218]:
```suggestion
  ## see https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/docs/best-practices.md#using-newer-kube-prometheus-stack.
```

**Comment by sumo-drosiek** [comment]
> I think we decided to put the new (kebab-case) docs in a separate directory at `/docs`, and leave the original file names in `/deploy/docs`. The original files in `/deploy/docs` should only contain links to the new files, e.g. `/deploy/docs/Best_Practices.md` should only contain one line like this:
> 
> ```
> Moved to [/docs/best-practices.md](../../docs/best-practices.md)
> ```

As markdown file should start with first level header `# ...` I will leave them and add redirection info below them

**Review by swiatekm-sumo — APPROVED** [review]
😰

**Pull request merged**